addons/project_timesheet_holidays/models/resource_calendar_leaves.py

```diff
@@ -143,6 +143,38 @@ def _timesheet_prepare_line_values(self, index, employee_id, work_hours_data, da
             'company_id': employee_id.company_id.id,
         }
 
+    def _generate_public_time_off_timesheets(self, employees):
+        timesheet_vals_list = []
+        work_hours_data = self._work_time_per_day()
+        timesheet_read_group = self.env['account.analytic.line'].read_group(
```

Comment:
We could use `_read_group` to order `many2one` field by id instead of the order defined in the comodel. 🙂 
```suggestion
        timesheet_read_group = self.env['account.analytic.line']._read_group(
```